portfolio/src/main/java/com/google/sps/servlets/SpotifyServlet.java

```diff
@@ -0,0 +1,150 @@
+package com.google.sps.servlets;
+
+import com.wrapper.spotify.SpotifyApi;
+import com.wrapper.spotify.SpotifyHttpManager;
+import com.wrapper.spotify.exceptions.SpotifyWebApiException;
+import com.wrapper.spotify.model_objects.credentials.AuthorizationCodeCredentials;
+import com.wrapper.spotify.model_objects.miscellaneous.CurrentlyPlaying;
+import com.wrapper.spotify.model_objects.specification.Track;
+import com.wrapper.spotify.model_objects.IPlaylistItem;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeRefreshRequest;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeRequest;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeUriRequest;
+import com.wrapper.spotify.requests.data.player.GetUsersCurrentlyPlayingTrackRequest;
+import org.apache.hc.core5.http.ParseException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/spotify")
+public class SpotifyServlet extends HttpServlet {
+
+  private static final String clientId = "444dda318d8949cb9282d95a9885a7c6";
+  private static final String clientSecret = "9f81b8a6cf5748ef9d24a7ddb1e8bc4c";
```

Comment:
I'm not entirely familiar with what this is but usually 'secret' keys like API keys are checked in to a separate file (could be as simple as a Java class that does nothing but has a few public properties) that is hidden by .gitignore. The repos then usually have instructions on how to make your own version of that file locally